### PR TITLE
Cleanup schema sampling procs

### LINKED ISSUE #1903: Cleanup apoc.meta.*TypeProperties()
Currently, the code for apoc.meta.nodeTypeProperties() and apoc.meta.relTypeProperties() can use some cleanup. Bonus: APOC does not compile under windows thanks to encoding errors. Will tackle both of these issues with a PR.

### PR DESCRIPTION
**Pull request by SaschaPeukert**
Fixes #1903

Cleanup apoc.meta.nodeTypeProperties() and apoc.meta.relTypeProperties()

## Proposed Changes

A brief list of proposed changes in order to fix the issue:

  - Make APOC compile on windows
  - Code cleanup

### CODE REVIEW

File: core/src/test/java/apoc/meta/MetaTest.java [hunk 1]
```diff
@@ -132,11 +127,11 @@ public static boolean testDBCallEquivalence(GraphDatabaseService db, String test
...
-        System.out.println("COMPARE TO:");
-        System.out.println(toCSV(compareTo.get()));
-
-        System.out.println("TEST SET:");
-        System.out.println(toCSV(testSet.get()));
+//        System.out.println("COMPARE TO:");
+//        System.out.println(toCSV(compareTo.get()));
+//
+//        System.out.println("TEST SET:");
+//        System.out.println(toCSV(testSet.get()));
```
Comment (Lojjs) [inline core/src/test/java/apoc/meta/MetaTest.java:134]: I think out-commented code is just slightly less bad than printlines in the code... Maybe at least add a comment saying something like `uncomment this to debug` so it is clear they are there on purpose.

File: core/src/test/java/apoc/meta/MetaTest.java [hunk 2]
```diff
@@ -808,198 +803,457 @@ public void testMetaGraphExtraRelsWithSample() throws Exception {
...
+                assertEquals( List.of( "Long", "String" ), rec.get( "propertyTypes" ) );
+                assertEquals( 2L, rec.get( "propertyObservations" ) );
+                assertEquals( 2L, rec.get( "totalObservations" ) );
+                assertEquals( false, rec.get( "mandatory" ) );
+            }
+        } );
+    }
+
+    @Test
+    public void testNodeTypePropertiesWithSpecialSampleSize()
```
Comment (Lojjs) [inline core/src/test/java/apoc/meta/MetaTest.java:1157]: Even if the behaviour is a bit odd in this cases I think it is nice with this test that show the current behaviour!

**Review by Lojjs — APPROVED** [review]
Good tests and cleanup!

File: core/src/main/java/apoc/meta/Meta.java [hunk 3]
```diff
@@ -498,40 +492,45 @@ private void collectStats(SubGraph subGraph, Collection<String> labelNames, Coll
...
-        try {
-            return collectTables4LabelsProfile(metaConfig).asRelStream();
-        } catch (Exception e) {
-            log.debug("meta.relTypeProperties(): Failed to return stream", e);
-            throw new RuntimeException(e);
+    @Description( "apoc.meta.relTypeProperties()" )
+    public Stream<Tables4LabelsProfile.RelTypePropertiesEntry> relTypeProperties( @Name( value = "config", defaultValue = "{}" ) Map<String,Object> config )
+    {
+        MetaConfig metaConfig = new MetaConfig( config );
+        try
```
Comment (fbiville) [inline core/src/main/java/apoc/meta/Meta.java:526]: nit: Neo4j formatting rules do not apply to the APOC project

**Review by fbiville — APPROVED** [review]

File: core/src/main/java/apoc/meta/Meta.java (hunk 3 above)
Comment (SaschaPeukert) [reply core/src/main/java/apoc/meta/Meta.java:526]: Question: Do you even have consistent formatting rules in APOC?

Comment (fbiville) [reply core/src/main/java/apoc/meta/Meta.java:526]: Screenshot 2021-05-28 at 17 45 46

Only 2 results are returned for this coding style, so I'd at least avoid extra spaces within parentheses and newlines before curling braces.

We just filed https://github.com/neo4j-contrib/neo4j-apoc-procedures/issues/1936 to make this clearer for future contributors.

**Pull request merged**